app/controllers/proxy_controller.rb

```diff
@@ -17,7 +17,7 @@ class ProxyController < ActiveStorage::BaseController
   # rubocop:disable Metrics/AbcSize
   def show
     attachment = ActiveStorage::Attachment.find(params[:id])
-    authorize!(:read, attachment.record)
+    authorize!(:read, attachment.record.stream)
```

Comment:
I think this should be legit given that all our records are associated w/ streams...